src/components/trade/Price.tsx

```diff
@@ -225,15 +229,17 @@ export const Price: React.FC<Props> = ({
     inputId: priceInverseInputId,
     precision: DEFAULT_PRECISION,
   })
-  const isDirect = priceShown === 'DIRECT'
+
+  const hideInverse = (!isPriceInverted && wasPriorityAdjusted) || (isPriceInverted && !wasPriorityAdjusted)
```

Comment:
This part is a bit strange, very complex to understand, minimum we need a comment to clarify.
Also, is not true that only one price will be visible. Therefore, only one const should be enought

Reply:
fair enough on 1 constant required

Reply:
I am going to rework thru the other comments to try and cut complexity but i still think the 2nd flag is required to balance the inverse/direct price with the act of changing the token order when getting pricing. Will add comment reflecting this